From fdaa00fb91f5434820427b3f8af571bfdcabb208 Mon Sep 17 00:00:00 2001
From: yeamour <ye_amour0203@126.com>
Date: Sat, 18 Apr 2015 20:34:28 +0800
Subject: [PATCH] Update test_enum.cs
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

测试thrift
---
 UnitTestDll/test_enum.cs | 25 +++++++++++++++++++++++++
 1 file changed, 25 insertions(+)

diff --git a/UnitTestDll/test_enum.cs b/UnitTestDll/test_enum.cs
index 30d401d..40f0093 100644
--- a/UnitTestDll/test_enum.cs
+++ b/UnitTestDll/test_enum.cs
@@ -91,6 +91,31 @@ static void UnitTest_13()
             bool[] data = new bool[4] { true,true,true,true};
             data[0] = false;
         }
+        
+        static void UnitTest_14_TestMemoryBuffer()
+        {
+            Logger.Log("Test Thrift UnitTest_14_TestMemoryBuffer Makr1");
+            Blog blog = new Blog();
+            blog.Content = new byte[] { 1 };
+            blog.CreatedTime = 2;
+            blog.Id = "TestBlog";
+            blog.IpAddress = "HereIsIpAddress";
+            blog.Topic = "TestTopic";
+            Dictionary<string, string> tmp = new Dictionary<string, string>();
+            tmp.Add("testKey", "testValue");
+            blog.Props = tmp;
+            Logger.Log("Test Thrift UnitTest_14_TestMemoryBuffer Makr2");
+            byte[] data = TMemoryBuffer.Serialize(blog);
+            Logger.Log("Test Thrift UnitTest_14_TestMemoryBuffer Makr3");
+
+            //Blog result = TMemoryBuffer.DeSerialize<Blog>(data);
+
+            TMemoryBuffer transport = new TMemoryBuffer(data);
+            TBinaryProtocol prot = new TBinaryProtocol(transport);
+            Blog result = new Blog();
+            result.Read(prot);
+            Console.WriteLine("Id " + result.Id + " IpAddress " + result.IpAddress + " Props.testKey " + result.Props["testKey"]);
+        }
 
     }
 
-- 
1.7.11.1

